shared/src/test/scala/mlscript/DiffTests.scala

```diff
@@ -537,6 +539,30 @@ class DiffTests extends org.scalatest.funsuite.AnyFunSuite with org.scalatest.Pa
             }
             // generate typescript typegen block
             if (mode.generateTsDeclarations) outputSourceCode(tsTypegenCodeBuilder.toSourceCode())
+
+            if (mode.showVariance) {
+              // process typer typedefs for type defintions
+              // in current block that have at least one
+              // type variable
+              val curBlockTypeDefs = typeDefs
+                .flatMap(typeDef => ctx.tyDefs.get(typeDef.nme.name))
+                .filter(tyDef => !tyDef.tparamsargs.isEmpty)
+              typer.computeVariances(curBlockTypeDefs, ctx)
+
+              val allSource = curBlockTypeDefs
+                .map(tyDef => {
+                  val nameSource = SourceCode(s"${tyDef.kind.str} ${tyDef.nme.name}")
+                  val typeDefSource = SourceCode.horizontalArray(tyDef.tparamsargs.map{case (t, tvar) =>
+                      SourceCode(s"${tyDef.tvarVariance.getOrElse(tvar, throw new Exception(s"Type variable $tvar not found in variance store"))}$tvar")
+                    }.toList)
+                  nameSource ++ typeDefSource
+                })
+              outputSourceCode(
```

Comment:
Actually, it would be better to always show the inferred variance alongside the "`Defined <kind> <name>`" messages – updating these messages to the format "`Defined <kind> <name>[<variance><type param name>*]`".